src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -274,7 +264,10 @@ def create_optimizer(self):
                 wrap_optim=self.scaler,
                 loggers=self.logger_manager,
                 distillation_teacher=self.teacher,
-                grad_sampler=self.grad_sampler,
+                grad_sampler={
```

Comment:
There's a type change here, make sure this is should be grad_sampler and not grad_sampler_kwargs

Reply:
Yes this should be a dictionary with keys: `data_loader_builder` and `loss_function`, as we now delegate instantiating `GradSampler` class to the pruner's `initialize` procedure. That way we make it possible to override some of the default gradient-sampling configs. For example in upstream BERT runs, we don't want to use the default training batchsize=256 to sample gradients for pruning, and we want to set this to something much smaller via yaml recipe.